tests/drawer/test_draw.py

```diff
@@ -281,77 +281,141 @@ def circ():
-
-    def circ(weights):
-        qml.RX(weights[0], 0)
-        qml.measure(0, reset=True)
-        qml.RX(weights[1], 1)
-        qml.measure(1)
-        qml.CNOT([0, 3])
-        qml.measure(3, postselect=0, reset=True)
-        qml.RY(weights[2], 2)
-        qml.CNOT([1, 2])
-        qml.measure(2, postselect=1)
-        qml.MultiRZ(0.5, [0, 2])
-        return qml.expval(qml.PauliZ(2))
-
-    drawing = qml.draw(circ)(np.array([np.pi, 3.124, 0.456]))
-    expected_drawing = (
-        "0: ──RX(3.14)──┤↗│  │0⟩─╭●─────────────────────╭MultiRZ(0.50)─┤     \n"
-        "1: ──RX(3.12)──┤↗├──────│─────────────╭●───────│──────────────┤     \n"
-        "3: ─────────────────────╰X──┤↗₀│  │0⟩─│────────│──────────────┤     \n"
-        "2: ──RY(0.46)─────────────────────────╰X──┤↗₁├─╰MultiRZ(0.50)─┤  <Z>"
-    )
+        def circ(phi):
+            qml.RX(phi, 0)
+            m0 = qml.measure(0)
+            qml.cond(m0, qml.PauliX)(wires=1)
+
+        drawing = qml.draw(circ)(np.pi)
+        expected_drawing = (
+            "0: ──RX(3.14)──┤↗├────┤  \n1: ─────────────║───X─┤  \n                ╚═══╝    "
+        )
+
+        assert drawing == expected_drawing
+
+    def test_single_meas_single_cond_multi_wire(self):
+        """Test that a multi-wire conditional operator relying on a single
+        mid-circuit measurement is drawn correctly."""
+
+        def circ(phi, theta):
+            qml.RX(phi, 0)
+            m0 = qml.measure(0)
+            qml.RY(theta, 2)
+            qml.cond(m0, qml.CNOT)(wires=[1, 0])
+
+        drawing = qml.draw(circ)(np.pi, np.pi / 2)
+        expected_drawing = (
+            "0: ──RX(3.14)──┤↗├───────────╭X─┤  \n"
+            "2: ─────────────║───RY(1.57)─│──┤  \n"
+            "1: ─────────────║────────────╰●─┤  \n"
+            "                ╚═════════════╝    "
+        )
+
+        assert drawing == expected_drawing
+
+    def test_single_meas_multi_cond(self):
+        """Test that a circuit where a single mid-circuit measurement is used for multiple
+        conditions is drawn correctly"""
+
+        def circ():
+            qml.RX(0.5, 0)
+            qml.RX(0.5, 1)
+            m0 = qml.measure(1, reset=True, postselect=1)
+            qml.cond(m0, qml.MultiControlledX)(wires=[1, 2, 0], control_values="10")
+            qml.CNOT([3, 2])
+            qml.cond(m0, qml.ctrl(qml.MultiRZ, control=[1, 2], control_values=[True, False]))(
+                0.5, wires=[0, 3]
+            )
+            qml.cond(m0, qml.PauliX)(0)
+            return qml.expval(qml.PauliZ(0))
+
+        drawing = qml.draw(circ)()
+        expected_drawing = (
+            "0: ──RX(0.50)────────────╭X────╭MultiRZ(0.50)──X─┤  <Z>\n"
+            "1: ──RX(0.50)──┤↗₁│  │0⟩─├●────├●──────────────║─┤     \n"
+            "2: ─────────────║────────╰○─╭X─├○──────────────║─┤     \n"
+            "3: ─────────────║─────────║─╰●─╰MultiRZ(0.50)──║─┤     \n"
+            "                ╚═════════╩═════╩══════════════╝       "
```

Comment:
Do we have a test for _without_ the postselection (since anyway postselection and conditioning is arguably not too useful)

Reply:
Thanks!